ergo/metaculus.py

```diff
@@ -698,67 +703,61 @@ def show_prediction(
         :param num_samples: number of samples from the community
         :param **kwargs: additional plotting parameters
         """
+        if plot not in ("samples", "fitted", "both"):
+            raise ValueError("side keyword must be either 'samples', 'fitted', 'both'")
 
-        if isinstance(samples, SubmissionMixtureParams):
-            prediction = samples
-            prediction_normed_samples = pd.Series(
-                [logistic.sample_mixture(prediction) for _ in range(0, num_samples)]
-            )
+        df = pd.DataFrame()
 
-        else:
+        if plot in ("samples", "both"):
             if isinstance(samples, list):
                 samples = pd.Series(samples)
-            if not type(samples) in [pd.Series, np.ndarray]:
+            if not type(samples) in [pd.DataFrame, pd.Series, np.ndarray]:
                 raise ValueError(
                     "Samples should be a list, numpy arrray or pandas series"
                 )
             num_samples = samples.shape[0]
-            prediction_normed_samples = self.normalize_samples(
-                samples
-            )  # need to do go to normalizes space for dates values samples
-
-        if show_community:
-            df = pd.DataFrame(
-                data={
-                    "community": [  # type: ignore
-                        self.sample_normalized_community()
-                        for _ in range(0, num_samples)
-                    ],
-                    "prediction": prediction_normed_samples,  # type: ignore
-                }
-            )
-            # get domain for graph given the percentage of distribution kept
-            _xmin, _xmax = self.denormalize_samples(
-                self.get_central_quantiles(
-                    prediction_normed_samples,
-                    percent_kept=percent_kept,
-                    side_cut_from=side_cut_from,
-                )
-            )
 
-            df["prediction"] = self.denormalize_samples(df["prediction"])
-            df["community"] = self.denormalize_samples(df["community"])
+            if type(samples) == pd.DataFrame:
+                if plot != "samples" and samples.shape[1] > 1:
+                    raise ValueError(
```

Comment:
I'm a bit worried about how complicated this is...

Reply:
That makes sense. The least I could do is comment it a bit more. Let's talk about the basic idea though. It has to be something like this if we want to have the samples param be a flexible thing that takes in a 1-d sample or 2-d (dataframe) of samples. The alternative would be that we could make it less magic-y and make it explicit for the user with a separate keyword param  "models_df" or something. I kind of like the magic-y version, but I don't feel very strongly about it.

Reply:
Yeah I agree that the magic-y version seems better.

I think the main thing I was reacting to is that you can't do `plot_fitted` for multiple predictions. But I guess that's reasonable enough, makes sense to leave it as is at least for now